build/Common.nonprod.props

```diff
@@ -29,6 +29,8 @@
     <MicrosoftExtensionsHostingPkgVer>8.0.1</MicrosoftExtensionsHostingPkgVer>
     <MicrosoftNETTestSdkPkgVer>[17.11.1,18.0)</MicrosoftNETTestSdkPkgVer>
     <OpenTelemetryExporterInMemoryPkgVer>$(OpenTelemetryCoreLatestVersion)</OpenTelemetryExporterInMemoryPkgVer>
+    <!-- OmniSharp/VS Code requires TargetFrameworks to be in descending order for IntelliSense and analysis. -->
+    <SupportedNetTargetsWithNet9>net9.0;net8.0</SupportedNetTargetsWithNet9>
     <SupportedNetTargets>net8.0</SupportedNetTargets>
```

Comment:
Can we just try to add to all packages by default? If it pass without issues, we could merge it. IMO it should be standard procedure on the repo-level instead of handling it one by one.

The only exception can occur when the tests are failing. Then we can create `SupportedNetTargetsWithoutNet9` to handle is separately.

Reply:
There were a lot of issues in this PR and I already split into another PR (#2316).

Can I keep this PR simple and do the follow up for other packages in another PR?

Reply:
I have added `net9.0` for all test projects except for `OpenTelemetry.Instrumentation.StackExchangeRedis.Tests` and `OpenTelemetry.Resources.AWS.Tests` because the tests are failing. `SupportedNetTargetsWithoutNet9` is added for them.